.github/workflows/docker_job.yml

```diff
@@ -36,12 +42,15 @@ jobs:
       - name: Upload Trivy scan results to GitHub Security tab
         id: trivy_upload_sarif
         uses: github/codeql-action/upload-sarif@v2
-        if: always()
+        if: inputs.changes_detected == true
```

Comment:
you might need to do always() && inputs.changes_detected == true here. If you do this then it won't try and run if changes not detected but will still always run even on failure of previous job. If I remember rightly why it was set to always() in the first place.

Reply:
that's gone back to always()